pkg/tools/doc/types.go

```diff
@@ -0,0 +1,307 @@
+	XKclDictKeyType *KclType        `json:"x-kcl-dict-key-type,omitempty"` // dict key type
+}
+
+// XKclModelType defines the `x-kcl-type` extension
+type XKclModelType struct {
+	Type   string              `json:"type,omitempty"`   // schema short name
+	Import *KclModelImportInfo `json:"import,omitempty"` // import information
+}
+
+// KclModelImportInfo defines how to import the current type
+type KclModelImportInfo struct {
+	Package string `json:"package,omitempty"` // import package path
+	Alias   string `json:"alias,omitempty"`   // import alias
+}
+
+// XKclDecorators defines the `x-kcl-decorators` extension
+type XKclDecorators struct {
+	Name      string
+	Arguments []string
+	Keywords  map[string]string
+}
+
+// GetKclTypeName get the string representation of a KclType
+func (tpe *KclType) GetKclTypeName(omitAny bool) string {
+	if tpe.Ref != "" {
+		return tpe.Ref[strings.LastIndex(tpe.Ref, ".")+1:]
+	}
+	switch tpe.Type {
+	case String:
+		if tpe.ReadOnly {
+			return tpe.Default
+		}
+		return string(Str)
+	case Integer:
+		if tpe.Format == Int64 {
+			if tpe.ReadOnly {
+				return tpe.Default
+			}
+			return string(Int)
+		}
+		panic(fmt.Errorf("unexpected KCL OpenAPI type and format: %s(%s)", tpe.Type, tpe.Format))
+	case Number:
+		if tpe.Format == Float {
+			if tpe.ReadOnly {
+				return tpe.Default
+			}
+			return string(FloatKcl)
+		}
+		panic(fmt.Errorf("unexpected KCL OpenAPI type and format: %s(%s)", tpe.Type, tpe.Format))
+	case Bool:
+		if tpe.ReadOnly {
+			return tpe.Default
+		}
+		return string(BoolKcl)
+	case Array:
+		return fmt.Sprintf("[%s]", tpe.Items.GetKclTypeName(true))
+	case Object:
+		if tpe.AdditionalProperties != nil {
+			// dict type
+			if tpe.KclExtensions.XKclDictKeyType.isAnyType() && tpe.AdditionalProperties.isAnyType() {
+				return "{}"
+			}
+			return fmt.Sprintf("{%s:%s}", tpe.KclExtensions.XKclDictKeyType.GetKclTypeName(true), tpe.AdditionalProperties.GetKclTypeName(true))
+		}
+		if tpe.KclExtensions != nil && len(tpe.KclExtensions.XKclUnionTypes) > 0 {
+			// union type
+			tpes := make([]string, len(tpe.KclExtensions.XKclUnionTypes))
+			for i, unionType := range tpe.KclExtensions.XKclUnionTypes {
+				tpes[i] = unionType.GetKclTypeName(true)
+			}
+			return strings.Join(tpes, " | ")
+		}
+		if tpe.isAnyType() {
+			if omitAny {
+				return ""
+			} else {
+				return "any"
+			}
+		}
+	default:
+		panic(fmt.Errorf("unexpected KCL OpenAPI type: %s", tpe.Type))
+	}
+	return string(tpe.Type)
+}
+
+// isAnyType checks if a KclType is any type
+func (tpe *KclType) isAnyType() bool {
+	return tpe.Type == Object && tpe.Properties == nil && tpe.AdditionalProperties == nil && tpe.Ref == "" && (tpe.KclExtensions == nil || tpe.KclExtensions.XKclUnionTypes == nil)
+}
+
+// GetKclOpenAPIType converts the kcl.KclType(the representation of Type in KCL API) to KclType(the representation of Type in KCL Open API)
+func GetKclOpenAPIType(from *kcl.KclType, defs map[string]*KclType) *KclType {
```

Comment:
For the return type, use `KclOpenAPIType` instead of `KclType` to differentiate KclType.

Reply:
fixed, ptal